app/models/resource.rb

```diff
@@ -8,4 +12,41 @@ class Resource < ActiveRecord::Base
   has_many :services
   has_many :ratings
   has_many :change_requests
+
+  if Rails.configuration.x.algolia.enabled
```

Comment:
Right, I had the same train of thought. I figured it'd probably be better give in to the way `algolia` expects you to integrate it, since it's less error-prone.